src/module/rules/rule-element/crit-spec.ts

```diff
@@ -1,29 +1,85 @@
             alternate: new fields.BooleanField(),
             text: new fields.StringField({ blank: false, nullable: true, initial: null }),
+            damageDice: new fields.SchemaField(
+                {
+                    number: new ResolvableValueField({ required: true, nullable: false, initial: undefined }),
+                    faces: new fields.NumberField({
+                        required: true,
+                        nullable: false,
+                        choices: [4, 6, 8, 10, 12],
+                        initial: undefined,
+                    }),
+                    damageType: new fields.StringField({
+                        required: false,
+                        nullable: true,
+                        choices: () => CONFIG.PF2E.damageTypes,
+                        initial: null,
+                    }),
+                    category: new fields.StringField({
+                        required: false,
+                        nullable: true,
+                        choices: Array.from(DAMAGE_CATEGORIES_UNIQUE),
+                        initial: null,
+                    }),
+                },
+                { required: false, nullable: true, initial: null }
+            ),
+            modifier: new fields.SchemaField(
+                {
+                    type: new fields.StringField({
+                        required: true,
+                        nullable: false,
+                        choices: Array.from(MODIFIER_TYPES),
+                        initial: "untyped",
+                    }),
+                    damageType: new fields.StringField({
+                        required: false,
+                        nullable: true,
+                        choices: () => CONFIG.PF2E.damageTypes,
+                        initial: null,
+                    }),
+                    category: new fields.StringField({
+                        required: false,
+                        nullable: true,
+                        choices: Array.from(DAMAGE_CATEGORIES_UNIQUE),
+                        initial: null,
+                    }),
+                    value: new ResolvableValueField({ required: true, nullable: false, initial: undefined }),
+                },
+                { required: false, nullable: true, initial: null }
+            ),
         };
     }
 
     static override validateJoint(data: SourceFromSchema<CritSpecRuleSchema>): void {
         super.validateJoint(data);
-        if (data.alternate && !data.text) {
-            throw Error("  alternate: must also include text");
+
+        if (data.alternate && !data.text && !data.damageDice && !data.modifier) {
+            throw Error("  alternate: must also include at least one of text, damage dice, or modifier");
+        } else if (!data.alternate && (data.text || data.damageDice || data.modifier)) {
+            const badProperty = data.text ? "text" : data.damageDice ? "damageDice" : data.modifier;
```

Comment:
Final ternary case probably shouldn't be data.modifier but rather "modifier".

Why not `const badProperty = ["text", "damageDice", "modifier"].find((p) => !!data[p])`?